### PR TITLE
Add option to show number of selected tracks in status bar

### LINKED ISSUE #315: Suggestion - Show number of selected items in status bar
I miss the feature to show the number of selected items in the status bar.

I am currently compiling my own plugin and hijack the "Shot total length of selected tracks" option with the following diff:
``` diff
diff --git a/foo_ui_columns/status_bar.cpp b/foo_ui_columns/status_bar.cpp
index 7987682..ddb4730 100644
--- a/foo_ui_columns/status_bar.cpp
+++ b/foo_ui_columns/status_bar.cpp
@@ -190,6 +190,12 @@ void get_selected_length_text(pfc::string_base& p_out, unsigned dp = 0)
     length = sels.calc_total_duration();

     p_out = pfc::format_time_ex(length, dp);
+       p_out.add_string("    ");
+       p_out.add_string(pfc::format_int(count));
+       p_out.add_string(" track");
+       if (count != 1) {
+               p_out.add_string("s");
+       }
 }
```

image

This is a bit crude - a separate option for it would be much nicer.

But this would mean adding new preferences / ui elements / adapt size calculation, which are not that straight-forward as a non-c++ developer.

### PR DESCRIPTION
**Pull request by m0vie**
Fixes #315.

image

image

### CODE REVIEW

**Review by reupen — COMMENTED** [review]
This looks great! Thanks very much.

One other comment:

- the `cui::status_bar::set_part_sizes()` call in `main_window::config_set_status_show_lock()` also needs updating to pass `cui::status_bar::t_part_count` (otherwise the parts get messed up when toggling the playlist lock part when on a locked playlist)

(Feel free to amend the existing commit to save time squashing later.)

File: foo_ui_columns/status_bar.cpp [hunk 1]
```diff
@@ -148,6 +149,12 @@ int calculate_selected_length_size(const char* p_text)
         win32_helpers::status_bar_get_text_width(g_status, "0d 00:00:00"));
 }
 
+int calculate_selected_count_size(const char* p_text)
+{
+    return std::max(win32_helpers::status_bar_get_text_width(g_status, p_text),
+        win32_helpers::status_bar_get_text_width(g_status, "00 tracks"));
```
Comment (reupen) [inline foo_ui_columns/status_bar.cpp:155]: I think I'd bump this up in size a bit as it seems a bit inconsistent compared to the length of the selected tracks:

```suggestion
        win32_helpers::status_bar_get_text_width(g_status, "1,000 tracks"));
```

File: foo_ui_columns/foo_ui_columns.rc [hunk 2]
```diff
@@ -463,8 +463,9 @@ BEGIN
     CONTROL         "Show status bar",IDC_SHOW_STATUS,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,7,176,68,10
     CONTROL         "Show playlist lock status",IDC_SHOW_LOCK,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,7,194,96,10
     CONTROL         "Show total length of selected tracks",IDC_SELTIME,
-                    "Button",BS_AUTOCHECKBOX | WS_TABSTOP,159,176,134,10
-    CONTROL         "Show volume",IDC_VOL,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,159,194,61,10
+                    "Button",BS_AUTOCHECKBOX | WS_TABSTOP,113,176,134,10
+    CONTROL         "Show number of selected tracks",IDC_SELCOUNT,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,113,189,123,18
```
[the remark this answers is not in the document]
Comment (m0vie) [reply foo_ui_columns/foo_ui_columns.rc:467]: Done

File: foo_ui_columns/status_bar.cpp [hunk 3]
```diff
@@ -171,6 +178,15 @@ std::string get_selected_length_text(unsigned dp = 0)
     return pfc::format_time_ex(length, dp).get_ptr();
 }
 
+std::string get_selected_count_text(unsigned dp = 0)
+{
+    static_api_ptr_t<playlist_manager> playlist_api;
+
+    unsigned count = playlist_api->activeplaylist_get_selection_count(pfc_infinite);
+
+    return fmt::format("{:d} {}", count, count == 1 ? "track" : "tracks");
```
[the remark this answers is not in the document]
Comment (m0vie) [reply foo_ui_columns/status_bar.cpp:187]: Done. Good idea, haven't thought of that.

File: foo_ui_columns/config_vars.h [hunk 4]
```diff
@@ -57,7 +57,7 @@ extern ConfigMenuItem cfg_playlist_double;
 
 extern cfg_int cfg_global, cfg_cur_prefs_col, cfg_header_hottrack, cfg_sortsel, cfg_global_sort, cfg_vis, cfg_vis2,
     cfg_vis_edge, cfg_lock, cfg_header, cfg_drop_at_end, cfg_mclick, cfg_mclick2, cfg_balloon, cfg_scroll_h_no_v,
-    cfg_ellipsis, cfg_tabs_multiline, cfg_frame, cfg_show_seltime, cfg_plistframe, cfg_tooltips_clipped, cfg_np,
+    cfg_ellipsis, cfg_tabs_multiline, cfg_frame, cfg_show_seltime, cfg_show_selcount, cfg_plistframe, cfg_tooltips_clipped, cfg_np,
```
[the remark this answers is not in the document]
Comment (m0vie) [reply foo_ui_columns/config_vars.h:60]: I wasn't sure whether your workflow was to keep diffs for semantic changes as minimal as possible and from time to time do format-only commits vs. format every time even if it causes a larger diff.

Formatting should be fixed now.

File: foo_ui_columns/status_bar.cpp (hunk 1 above)
Comment (m0vie) [reply foo_ui_columns/status_bar.cpp:155]: Done

File: foo_ui_columns/config_vars.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (reupen) [reply foo_ui_columns/config_vars.h]: Thanks! It's a horrible line of code really (it's legacy stuff).

**Review by reupen — APPROVED** [review]

**Comment by reupen** [comment]
Thanks again!

**Pull request merged**